dotCMS/src/main/java/com/dotcms/contenttype/business/FieldAPIImpl.java

```diff
@@ -471,11 +473,64 @@ public FieldVariable save(final FieldVariable var, final User user) throws DotDa
       
       //update Content Type mod_date to detect the changes done on the field variables
       contentTypeAPI.updateModDate(type);
+
+      //Validates custom mapping format
+      if (var.key().equals(FieldVariable.ES_CUSTOM_MAPPING_KEY)){
+          try {
+              new JSONObject(var.value());
+          } catch (JSONException e) {
+              handleInvalidCustomMappingError(var, user, field, type, e);
+          }
+
+          //Verifies the field is marked as System Indexed. In case it isn't, the field will be updated with this flag on
+          if (!field.indexed()) {
+              save(FieldBuilder.builder(field).indexed(true).build(), user);
+              Logger.info(this, "Field " + type.variable() + "." + field.variable()
+                      + " has been marked as System Indexed as it has defined a field variable with key "
+                      + FieldVariable.ES_CUSTOM_MAPPING_KEY);
+          }
+      }
       
       return newFieldVariable;
   }
 
-  @Override
+    /**
+     *
+     * @param fieldVariable
+     * @param user
+     * @param field
+     * @param type
+     * @param exception
+     */
+    private void handleInvalidCustomMappingError(final FieldVariable fieldVariable, final User user,
+            final Field field, final ContentType type, final JSONException exception) {
+        final String[] messageParameters = new String[]{fieldVariable.key(), field.name(), type.name()};
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4633610)